### PR TITLE
Bootstrap an OpenShift conventions document for the project

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+This document identifies conventions adopted by the OpenShift project
+as a whole. Generally the conventions outlined below should be followed by all
+components that make up the project, but exceptions are allowed when necessary.
+
+
+## Motivation
+
+The conventions are intended to help with consistency across the project. Users
+of the platform expect this consistency in the experience and operation of the cluster.
```
Comment (smarterclayton) [inline CONVENTIONS.md:47]: Developers of the platform expect this consistency in the code to quickly identify issues across codebases.

Maybe a comment too about how consistency simplifies explanation, reduces accidental complexity.

**Comment by jwforres** [comment]
cc @derekwaynecarr @markmc @crawford @eparis @russellb since your names are on it as authors / approvers / reviewers

**Comment by derekwaynecarr** [comment]
we can add this as a follow-on PR, but I want to capture outcomes for some discussions.

When should an operand tolerate all taints?
1. if the operand is required to form a functional Kubernetes node
1. if the operand is required to support workloads sourced from an internal or external registry that core components depend upon

/cc @Miciah - i think this covers what we discussed.

**Comment by jwforres** [comment]
@derekwaynecarr yeah lets do that in a follow-on, it will need a new section

**Pull request merged**